### PR TITLE
Added some limits in 2x

### PR DESCRIPTION
**Pull request by shargon**
Close https://github.com/neo-project/neo/issues/1375

Ensure that MerkleBlockPayload fit the expected values

* Added small changes without relevance

### CODE REVIEW

File: neo/Ledger/ContractState.cs [hunk 1]
```diff
@@ -57,15 +56,15 @@ ContractState ICloneable<ContractState>.Clone()
...
             ContractProperties = (ContractPropertyState)reader.ReadByte();
-            Name = reader.ReadVarString();
-            CodeVersion = reader.ReadVarString();
-            Author = reader.ReadVarString();
-            Email = reader.ReadVarString();
-            Description = reader.ReadVarString();
+            Name = reader.ReadVarString(252);
+            CodeVersion = reader.ReadVarString(252);
+            Author = reader.ReadVarString(252);
+            Email = reader.ReadVarString(252);
```
Comment (Tommo-L) [inline neo/Ledger/ContractState.cs]: why 252?  Is 255 better, like the default varchar length in mysql?

Comment (Tommo-L) [reply neo/Ledger/ContractState.cs]: Got, the remaining 4 bytes represent the var length

Comment (shargon) [reply neo/Ledger/ContractState.cs]: The syscall has this same values

Comment (Tommo-L) [reply neo/Ledger/ContractState.cs]: 👍, thanks

**Comment by shargon** [comment]
@superboyiii could you test the backward compatibility?

**Comment by nicolegys** [comment]
> @superboyiii could you test the backward compatibility?

I'll test it, need about 2-3 days for chain sync.

**Comment by nicolegys** [comment]
TestResults:
Mainnet: compatible with 2.10.3
Testnet: 7 files different with 2.10.3/master2.x

[Differences_pr1376.zip](https://github.com/neo-project/neo/files/4029852/Differences_pr1376.zip)

It is strange that this pr is compatible with pr 1369 both on mainnet and testnet.

**Comment by shargon** [comment]
Are the same differences than https://github.com/neo-project/neo/pull/1369#issuecomment-571510853 ? seems very strange to me :S

**Comment by nicolegys** [comment]
> Are the same differences than [#1369 (comment)](https://github.com/neo-project/neo/pull/1369#issuecomment-571510853) ? seems very strange to me :S

Yes, I don't know why... Actually it's very strange to me, too.

**Comment by nicolegys** [comment]
@shargon Will you go on with this pr or close it?

**Comment by shargon** [comment]
I think that should be merged @nicolegys

**Comment by nicolegys** [comment]
Compatibility test pass. See reason on https://github.com/neo-project/neo/pull/1369#issuecomment-582331691

**Comment by Tommo-L** [comment]
Ready to merge?

**Pull request merged**